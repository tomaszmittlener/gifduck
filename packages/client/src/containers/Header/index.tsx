import React, { FunctionComponent } from 'react'
import styled from 'styled-components'
import Logo from 'components/Logo'

const Container = styled.header`
  position: sticky;
  top: 0;
  display: block;
  width: 100%;
  z-index: ${({ theme: { layers } }) => layers.top};
  background-color: ${({ theme }) => theme.colors.grays.black};
`

const SearchForm = styled.form`
  display: flex;
`

const SearchButton = styled.button`
  height: ${({ theme }) => theme.ms(10)};
  width: ${({ theme }) => theme.ms(10)};
  border: none;
  background-color: ${({ theme }) => theme.colors.grays.white};
  padding: 0;
  margin: 0;
  cursor: pointer;
`

const SearchInput = styled.input`
  border: none;
  line-height: 1.75;
  height: ${({ theme }) => theme.ms(10)};
  color: ${({ theme }) => theme.colors.primary.default};
  background-color: ${({ theme }) => theme.colors.grays.white};
  font-size: ${({ theme }) => theme.ms(0)};
  font-family: ${({ theme }) => theme.typography.fonts.default};
  padding: 0 ${({ theme }) => theme.ms(0)};
  width: 100%;
`

const Header: FunctionComponent = () => {
  return (
    <Container>
      <Logo />
      <SearchForm>
        <SearchInput />
        <SearchButton>🔎</SearchButton>
      </SearchForm>
    </Container>
  )
}

export default Header
