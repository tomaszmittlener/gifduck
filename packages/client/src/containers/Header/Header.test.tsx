import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'
import { MemoryRouter } from 'react-router-dom'

import Header from './'

const defaultProps = {
  className: 'class-name',
  testId: 'test-id',
}

afterEach(cleanup)

describe('Header', () => {
  test('renders correctly', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('passes className to container', async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <MemoryRouter>
          <Header {...defaultProps} />
        </MemoryRouter>
      </ThemeProvider>,
    )

    expect(getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  })
})
