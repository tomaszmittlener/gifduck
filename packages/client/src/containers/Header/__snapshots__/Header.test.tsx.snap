// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c2 {
  font-size: 2.2806973457336426rem;
  font-family: Pacifico,sans-serif;
  color: #F2C94C;
  line-height: 2;
  width: auto;
  -webkit-animation: dGzoaH 1s ease-out none;
  animation: dGzoaH 1s ease-out none;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  width: 100%;
  z-index: 1500;
  background-color: #343339;
}

.c1 {
  padding: 0 1rem;
  grid-column-gap: 0.49327018427257213rem;
  display: grid;
  grid-template-columns: repeat(8,1fr);
  width: 100%;
  max-width: 1440px;
  margin: 0 auto;
}

.c4 {
  grid-column: 1 / 9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 0 1.601806640625rem 0;
}

.c6 {
  height: 3.247321025468409rem;
  width: 3.247321025468409rem;
  border: none;
  background-color: #ECE9DE;
  padding: 0;
  margin: 0;
  cursor: pointer;
  border-radius: 0 10px 10px 0;
}

.c5 {
  border: none;
  line-height: 1.75;
  height: 3.247321025468409rem;
  color: #5D899E;
  background-color: #ECE9DE;
  font-size: 1rem;
  font-family: Open Sans,sans-serif;
  padding: 0 1rem;
  width: 100%;
  border-radius: 10px 0 0 10px;
}

.c3 {
  padding: 0.7901234567901234rem 0;
  grid-column: 1 / 9;
  justify-self: center;
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 0 1.423828125rem;
  }
}

@media screen and (min-width:1024px) {
  .c1 {
    padding: 0 2.0272865295410156rem;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    grid-column-gap: 1rem;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    justify-self: left;
    grid-column: 1 / 4;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        üê§gifduck
      </label>
      <form
        class="c4"
      >
        <input
          class="c5"
          name="search"
          placeholder="Search for Images..."
          type="text"
        />
        <button
          class="c6"
        >
          üîé
        </button>
      </form>
    </div>
  </header>
</div>
`;
